### PR TITLE
Canonicalize APIs exported by pkg:test_api

### PR DESCRIPTION
**Pull request by kevmoo**
Also fix a bunch of spelling throughout the monorepo

### CODE REVIEW

File: pkgs/test_api/lib/scaffolding.dart
Comment (kevmoo) [inline pkgs/test_api/lib/scaffolding.dart:5]: image

**Comment by natebosch** [comment]
> Also fix a bunch of spelling throughout the monorepo

Can this go out separately? Spelling fixes will be easy to approve without much consideration and not likely to need rollback.

I think we need to pay close attention to the canonicalization choices so I'd prefer if this PR didn't have extra noise.

File: pkgs/test_api/lib/scaffolding.dart
Comment (natebosch) [reply pkgs/test_api/lib/scaffolding.dart:5]: I think we're still in a place where we expect most folks to get these through `package:test/test.dart`, and if they didn't get it from there I think we'd want them to use `package:test/scaffolding.dart`, not `package:test_api/scaffolding.dart`.

**Comment by kevmoo** [comment]
PTAL

File: pkgs/test_api/lib/scaffolding.dart
Comment (kevmoo) [reply pkgs/test_api/lib/scaffolding.dart:5]: We can put notes here saying such.

In the short term, this fixes the dartdoc warnings.

Comment (jakemac53) [reply pkgs/test_api/lib/scaffolding.dart:5]: Are there analyzer implications here? Will it start adding this import instead of `package:test/test.dart` or `package:test/scaffolding.dart`? It seems reasonable that it would, but then this isn't the behavior we want.

Comment (jakemac53) [reply pkgs/test_api/lib/scaffolding.dart:5]: Can you maybe clarify or point to some docs on exactly what this is/does?

Comment (kevmoo) [reply pkgs/test_api/lib/scaffolding.dart:5]: https://github.com/dart-lang/dartdoc/issues/2887

Sadly, there are no docs. Basically, if one member is exposed by more than one library, `dartdoc` guesses where to put it.

This "annotation" makes things explicit

Comment (jakemac53) [reply pkgs/test_api/lib/scaffolding.dart:5]: Ok, it sounds like this should be fairly low impact then and we could change it later if we decided to?

Comment (kevmoo) [reply pkgs/test_api/lib/scaffolding.dart:5]: yup

**Review by jakemac53 — APPROVED** [review]
Lgtm although might want to wait for approval from @natebosch as well.

**Review by natebosch — APPROVED** [review]

**Pull request merged**
